src/Repository/CreditRepository.php

```diff
@@ -0,0 +1,73 @@
+<?php
+namespace App\Repository;
+
+use Wikimedia\ToolforgeBundle\Service\ReplicasClient;
+
+class CreditRepository {
+
+	public function getDatabaseName( $domain, $family ) {
+		if ( $family === 'wikisource' ) {
+			if ( in_array( $domain, [ 'old', 'www', 'mul' ] ) ) {
+				$dbname = 'sourceswiki_p';
+			} elseif ( $domain === 'zh-min-nan' ) {
+				$dbname = 'zh_min_nanwikisource_p';
+			} else {
+				$dbname = $domain . $family . '_p';
+			}
+		} else {
+			$dbname = $domain . $family . '_p';
```

Comment:
A note that the `_p` isn't strictly needed here since ReplicasClient will do that for you, but you do otherwise normally need to append `_p`.